### PR TITLE
Avoid top-level `with ...;` in `pkgs/development/lisp*`

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

Using `with` at a top-level scope is [an anti-pattern](https://nix.dev/guides/best-practices#with-scopes). The tracking issue is #208242. This is a pure refactor: there is no functional change contained in this PR.

I tried out the other strategy of replacing top-level with in this PR. If symbol `foo` was pulled implicitly from attrset `pkgs`, I replaced `foo` with `pkgs.foo`. Prior to this change, this file was a mix of both styles (`foo` and `pkgs.foo`), and now it's just one.

There are no rebuilds when I run `nixpkgs-review rev HEAD`, so I've targeted master with this PR.

## Things done

- [x] Tested compilation of all packages that depend on this change using `nixpkgs-review rev HEAD`.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lisp-modules-new-obsolete/ql.nix [hunk 1]
```diff
@@ -1,99 +1,95 @@
...
+  inherit (pkgs) lib;
 
   # FIXME: automatically add nativeLibs based on conditions signalled
 
   # Try to keep this list sorted
   extras = {
     cffi-libffi = pkg: {
-      nativeBuildInputs = [ libffi ];
-      nativeLibs = [ libffi ];
+      nativeBuildInputs = [ pkgs.libffi ];
```
Comment (fricklerhandwerk) [inline pkgs/development/lisp-modules-new-obsolete/ql.nix:11]: Hm, I'd have done `extras = with pkgs; {` instead. Not sure this is worth the diff, as it's also adding noise now.

File: pkgs/development/lisp-modules-new-obsolete/ql.nix [hunk 2]
```diff
@@ -1,99 +1,95 @@
...
-      nativeLibs = [ cairo ];
+      nativeLibs = [ pkgs.cairo ];
     };
     cl-cairo2-xlib = pkg: {
-      nativeLibs = [ gtk2-x11 ];
+      nativeLibs = [ pkgs.gtk2-x11 ];
     };
     cl-devil = pkg: {
       nativeBuildInputs = [ pkgs.libdevil ];
       nativeLibs = [ pkgs.libdevil ];
```
Comment (philiptaron) [inline pkgs/development/lisp-modules-new-obsolete/ql.nix:48]: These lines are doing `pkgs.`

File: pkgs/development/lisp-modules-new-obsolete/ql.nix [hunk 3]
```diff
@@ -115,24 +111,24 @@ let
...
+      nativeLibs = [ pkgs.rabbitmq-c ];
     };
     cl-rdkafka = pkg: {
-      nativeBuildInputs = [ rdkafka ];
-      nativeLibs = [ rdkafka ];
+      nativeBuildInputs = [ pkgs.rdkafka ];
+      nativeLibs = [ pkgs.rdkafka ];
     };
     cl-readline = pkg: {
       nativeLibs = [ pkgs.readline ];
```
Comment (philiptaron) [inline pkgs/development/lisp-modules-new-obsolete/ql.nix:128]: More `pkgs.`

File: pkgs/development/lisp-modules-new-obsolete/ql.nix [hunk 4]
```diff
@@ -224,14 +220,14 @@ let
...
+  builtQlpkgs = lib.mapAttrs (n: v: build v) qlpkgs;
 
   build = pkg:
     let
       builtPkg = build-asdf-system pkg;
       withExtras = pkg //
-                   (optionalAttrs
-                     (hasAttr pkg.pname extras)
+                   (lib.optionalAttrs
+                     (lib.hasAttr pkg.pname extras)
```
Comment (philiptaron) [inline pkgs/development/lisp-modules-new-obsolete/ql.nix:230]: Here's the added `lib.` to see how it feels (vs. the tool-assisted extraction present in other PRs.)

**Review by philiptaron — COMMENTED** [review]
The thing I was going for in the PR (which was hand-crafted instead of tool-assisted) was _uniformity_, and in this case following the convention of roughly half this file.

I'd rather have one way or the other, not some mishmash.

**Comment by fricklerhandwerk** [comment]
I get that. I mean, you're doing a refactor, and I appreciate you're doing it conservatively, but in this case we could as well make an opinionated choice - the original motivation was increasing clarity and readability. And again, I think judicious use of `with` can help with that.

**Comment by philiptaron** [comment]
I'm sorry, I find a `with` scope going over 90% of the lines of the file, mixed in with use of `pkgs.` within that scope to not be code I wish to produce. I can't apply that opinionated feedback in this PR, as it's not an opinion I share.

The only place I find `with` to make sense is where the scope is minimal: a list, a small set like `meta`, the like. If it scrolls off the screen while reading the file, or won't be present in the diff because it's far away, it's effectively a top-level `with`. To me, _that's_ a judicious use.

You can see evidence of that confusion when code authors introduce smaller with scopes that were already covered by some larger one. `with lib` at the top scope, followed by `with lib` in the `meta` section. Use of `pkgs.` in a list when there's a top-level `with pkgs;`.

The other possible use of `with pkgs;` that I could have transformed in this PR is landing `with pkgs;` in front of every `nativeBuildInputs` or `buildInputs`. I chose not to do that since virtually every one of these lists contain just one member, and adding `pkgs.` is shorter.

**Comment by fricklerhandwerk** [comment]
Yeah, fair.

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by philiptaron** [comment]
This PR is obviated by #309696.

**Pull request closed without merging**